### ISSUE TITLE
Failed to populate memmap

### ISSUE THREAD

**Issue by ava57r** [qdrant/qdrant#5114]
Migration existing collection from v1.10.1 to v1.11.4

## Current Behavior
I see errors into log about memmap when I upgrade qdrant version from v1.10.1 to v1.11.4

## Steps to Reproduce
1. I have collection for testing updates v1.10.1
2. I just update bin qdrant to v1.11.4

## Context (Environment)

OS: Oracle Linux 8.6
rust version: rustc 1.81.0

LOG:

```
Sep 23 10:46:20 : INFO collection::shards::local_shard: Recovering collection test-test: 0/1 (0%)
Sep 23 10:46:20 : INFO collection::shards::local_shard: Recovered collection test-test: 1/1 (100%)
Sep 23 10:46:20 : WARN memory::mmap_ops: Failed to populate mmap "./storage/collections/test-test/3/segments/32bc911f-23ec-40a9-8f55-cf3cd33722fb/vector_index" to cache: Invalid argument (os error 22)
Sep 23 10:46:20 : WARN memory::mmap_ops: Failed to populate mmap "./storage/collections/test-test/3/segments/32bc911f-23ec-40a9-8f55-cf3cd33722fb/vector_storage/matrix.dat" to cache: Invalid argument (os error 22)
```

**Comment by ava57r** [comment]
OL8.6 has kernel `kernel-uek-5.4.17-2136.307.3`

>= OL8.7 has `kernel-uek-5.15.0-3.60.5.1`

**Comment by timvisee** [comment]
Thanks!

Would it be possible for you to update to OL8.7 and see if the issue persists? Or do you prefer to stick with OL8.6 instead?

**Comment by timvisee** [comment]
We just release [Qdrant 1.11.5](https://github.com/qdrant/qdrant/releases/tag/v1.11.5) with a hotfix for this problem.

Though it'll still show a warning message similar to what you see above, it'll fall back to populate the memmap pages using a different method.

We plan to release a proper fix a bit later, which will also remove the warnings.

**Comment by ava57r** [comment]
> Thanks!
> 
> Would it be possible for you to update to OL8.7 and see if the issue persists? Or do you prefer to stick with OL8.6 instead?

I upgraded it to OL8.7
Now It is working without warrings.

**Comment by timvisee** [comment]
Fantastic! Thanks for being flexible :smile:

**Comment by timvisee** [comment]
@ava57r It turns out to be a problem in OL8.6. It has extensions in the kernel that conflict with changes introduced in later Linux versions. OL8.7 fixes this problem, not breaking user space anymore.

We don't think it's worth to implement a workaround for an older version, especially because it got fixed in newer releases. In the very worst case you'd see the above mentioned warning, not introducing a significant problem.

Long story short: we recommend to upgrade to the newer version, which you of course, have already done.

Thanks again for the report!

**Comment by xzfc** [comment]
Relevant Oracle Linux issue: https://github.com/oracle/linux-uek/issues/23.
